src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -242,7 +242,45 @@ internal int SerializeLogRecord(LogRecord logRecord)
         var timestamp = logRecord.Timestamp;
         var cursor = 0;
         cursor = MessagePackSerializer.WriteArrayHeader(buffer, cursor, 3);
-        cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+
+        var categoryName = logRecord.CategoryName;
+        string eventName = null;
+
+        Span<byte> data = stackalloc byte[0];
+        int validNameLength = 0;
+
+        // If user configured explicit TableName, use it.
+        if (this.m_tableMappings != null && this.m_tableMappings.TryGetValue(categoryName, out eventName))
+        {
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (!this.shouldPassThruTableMappings)
+        {
+            eventName = this.m_defaultEventName;
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (categoryName.Length > 0)
+        {
+            int cursorStartIdx = cursor;
+            cursor = SanitizeCategoryName(buffer, cursor, ref cursorStartIdx, ref validNameLength, categoryName);
```

Comment:
You don't need to pass this argument to `SanitizeCategoryName`.

You could use this line: `int cursorStartIdx = cursor;` inside the `SanitizeCategoryName` method.

Reply:
Right! 
(assuming that there will be no additional logic added between Ln264 and Ln265.)